### PR TITLE
#87966 Import process document association

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    override fun supports(fileName: String) = fileName.matches(FILENAME_REGEX)
+
+    override fun import(request: ImportRequest) {
+        val formFlowKey = FILENAME_REGEX.matchEntire(request.fileName)!!.groupValues[1]
+        processDocumentDeploymentService.deploy(formFlowKey, request.content.toString(Charsets.UTF_8))
+    }
+
+    companion object {
+        private val FILENAME_REGEX = """config/process-document-link/(.*).json""".toRegex()
```
Comment (theo-ritense) [inline process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt]:
```suggestion
        private val FILENAME_REGEX = """config/process-document-link/([^/]*).json""".toRegex()
```

File: process-document/src/test/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporterTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Test
+    fun `should support process-document-link fileName`() {
+        assertThat(processDocumentLinkImporter.supports(FILENAME)).isTrue()
+    }
+
+    @Test
+    fun `should not support non-process-document-link fileName`() {
+        assertThat(processDocumentLinkImporter.supports("config/non-process-document-link/test.json")).isFalse()
```
Comment (theo-ritense) [inline process-document/src/test/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporterTest.kt]:
```suggestion
        assertThat(processDocumentLinkImporter.supports("config/process-document-link/aa/test.json")).isFalse()
```

File: process-document/src/main/java/com/ritense/processdocument/service/impl/CamundaProcessJsonSchemaDocumentDeploymentService.java [hunk 3]
```diff
@@ -84,17 +84,21 @@ public void deployChangedProcessDocumentLinks() throws IOException {
...
-                }
+                final var content = IOUtils.toString(resource.getInputStream(), UTF_8);
+                deploy(documentDefinitionName, content);
             } catch (IOException e) {
                 logger.error("Error while deploying process-document-link", e);
             }
         }
     }
 
+    public void deploy(String documentDefinitionName, String content) throws JsonProcessingException {
```
Comment (theo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/service/impl/CamundaProcessJsonSchemaDocumentDeploymentService.java:95]: Can you add an integration test for this new method? With the unit test of the importer, we trust on this method to work properly.

File: process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+class ProcessDocumentLinkImporter(
+    private val processDocumentDeploymentService: ProcessDocumentDeploymentService,
+) : Importer {
+
+    override fun type() = "processdocumentlink"
+
+    override fun dependsOn() = setOf("documentdefinition", "processdefinition")
```
Comment (theo-ritense) [inline process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt:31]: The application will not start up if these dependencies do not exist. Which they don't (yet)

File: process-document/src/main/java/com/ritense/processdocument/service/impl/CamundaProcessJsonSchemaDocumentDeploymentService.java (hunk 3 above)
Comment (Klaas-Ritense) [reply process-document/src/main/java/com/ritense/processdocument/service/impl/CamundaProcessJsonSchemaDocumentDeploymentService.java:95]: Added

File: process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    override fun supports(fileName: String) = fileName.matches(FILENAME_REGEX)
+
+    override fun import(request: ImportRequest) {
+        val formFlowKey = FILENAME_REGEX.matchEntire(request.fileName)!!.groupValues[1]
+        processDocumentDeploymentService.deploy(formFlowKey, request.content.toString(Charsets.UTF_8))
+    }
+
+    companion object {
+        private val FILENAME_REGEX = """config/process-document-link/([^/]*).json""".toRegex()
```
Comment (theo-ritense) [inline process-document/src/main/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporter.kt]:
```suggestion
        private val FILENAME_REGEX = """config/process-document-link/([^/]*)\.json""".toRegex()
```

File: process-document/src/test/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporterTest.kt [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Test
+    fun `should support process-document-link fileName`() {
+        assertThat(processDocumentLinkImporter.supports(FILENAME)).isTrue()
+    }
+
+    @Test
+    fun `should not support non-process-document-link fileName`() {
+        assertThat(processDocumentLinkImporter.supports("config/process-document-link/aa/test.json")).isFalse()
+    }
```
Comment (theo-ritense) [inline process-document/src/test/kotlin/com/ritense/processdocument/importer/ProcessDocumentLinkImporterTest.kt:60]:
```suggestion
        assertThat(processDocumentLinkImporter.supports("config/process-document-link/test-json")).isFalse()
    }
```

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**